### PR TITLE
NAS-123305 / 24.04 / Add spare support in draid

### PR DESCRIPTION
**Pull request by Qubad786**
## Context

Support for specifying data disks and spare disks has been added for a DRAID based vdev along with necessary validation required for it to function properly. Integration tests have also been updated to account for the changes so we can properly verify that the changes work as desired.

### CODE REVIEW

File: src/middlewared/middlewared/plugins/pool_/pool.py [hunk 1]
```diff
@@ -307,6 +315,8 @@ def disk_to_stripe(topology_type):
                         # normalize it so that it reflects the parity as well i.e DRAID1, DRAID2, etc.
                         # sample value of name here is: draid1:1d:2c:0s-0
                         entry['type'] = f'{i["type"]}{i["name"][len("draid"):len("draid") + 1]}'
+                        entry['draid_spare_disks'] = int(RE_SPARE.findall(i['name'])[0][1: -1])
```
Comment (yocalebo) [inline src/middlewared/middlewared/plugins/pool_/pool.py:318]:
```suggestion
                        entry['draid_spare_disks'] = int(RE_SPARE.findall(i['name'])[0][1:-1])
```

File: src/middlewared/middlewared/plugins/pool_/pool.py [hunk 2]
```diff
@@ -307,6 +315,8 @@ def disk_to_stripe(topology_type):
...
+                        entry['draid_data_disks'] = int(RE_DATA.findall(i['name'])[0][1: -1])
```
Comment (yocalebo) [inline src/middlewared/middlewared/plugins/pool_/pool.py:319]:
```suggestion
                        entry['draid_data_disks'] = int(RE_DATA.findall(i['name'])[0][1:-1])
```

**Review by yocalebo — APPROVED** [review]
Great job, I made a suggestion to remove the white-space surrounding the regex string splicing. Other than that, looks good. Thanks!

**Comment by themylogin** [comment]
backport

**Pull request merged**